espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/drivers/ComposeDriver.kt

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.appium.espressoserver.lib.drivers
+
+import androidx.compose.ui.semantics.SemanticsProperties
+import androidx.compose.ui.test.*
+import io.appium.espressoserver.lib.handlers.exceptions.InvalidElementStateException
+import io.appium.espressoserver.lib.handlers.exceptions.NoSuchElementException
+import io.appium.espressoserver.lib.helpers.findAllComposeNodeInteractions
+import io.appium.espressoserver.lib.helpers.getNodeInteractionById
+import io.appium.espressoserver.lib.model.*
+
+class ComposeDriver : AppDriver {
+    override val name = DriverContext.StrategyType.COMPOSE
+
+    override fun findElement(params: Locator): BaseElement {
+        val parentNodeInteraction =
+            params.elementId?.let { getNodeInteractionById(it) }
+        val nodeInteractions =
+            findAllComposeNodeInteractions(parentNodeInteraction, params)
+
+        if (nodeInteractions.fetchSemanticsNodes().isEmpty()) {
+            throw NoSuchElementException(
+                String.format(
+                    "Could not find compose element with strategy %s and selector %s",
+                    params.using, params.value
+                )
+            )
+        }
+        return ComposeElement(nodeInteractions[0])
+    }
+
+
+    override fun click(params: AppiumParams): Void? {
+        val nodeInteraction = getNodeInteractionById(params.elementId)
+        nodeInteraction.performClick()
+        return null
+    }
+
+    override fun getText(params: AppiumParams): String {
+        val nodeInteraction = getNodeInteractionById(params.elementId)
+        try {
+            return nodeInteraction.fetchSemanticsNode().config[SemanticsProperties.Text][0].text
```

Comment:
`fetchSemanticsNode` would throw an exception if no matches are found. See the comment above.
I would create an extension method for  SemanticsNodeInteraction, so then we could handle such situation properly, for example:

```
fun  SemanticsNodeInteraction.findMatch(): SemanticsNode =
  try {
    this.fetchSemanticsNode()
  } catch (e: AssertionError) {
     throw StaleElementError...
  }
```

Reply:
hm, to raise StaleElementError we will need to pass `elementId` to this extension method. How about a global function:
```
fun semanticsNode(elementId: String) =
    try {
        getNodeInteractionById(elementId).fetchSemanticsNode()
    } catch (e: AssertionError) {
        throw StaleElementException(elementId)
    }
```

Reply:
works for me. It is just a common practice to have a verb in function name. So something like `findMatchingSemanticNode` would probably fir better.

Reply:
I had changed it to `getSemanticsNode`